### PR TITLE
Experimental

### PR DESCRIPTION
**Pull request by hntd187**

### CODE REVIEW

File: src/cluster/rpc_server.rs [hunk 1]
```diff
@@ -65,7 +66,7 @@ impl RpcServer {
     pub fn create_client(uri: http::Uri) -> impl Future<Item = RpcClient, Error = ConnectError<Error>> {
         info!("Creating Client to: {:?}", uri);
         let dst = Destination::try_from_uri(uri.clone()).unwrap();
-        let connector = Connector::new(HttpConnector::new(num_cpus::get()));
+        let connector = Connector::new(HttpConnector::new(8));
```
Comment (LucioFranco) [inline src/cluster/rpc_server.rs]: you probably dont need 8

File: toshi-client/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,116 @@
+use std::fmt;
+
+use isahc::prelude::*;
+use isahc::HttpClientBuilder;
```
Comment (LucioFranco) [inline toshi-client/src/lib.rs]: why this and not hyper?

File: src/cluster/rpc_server.rs (hunk 1 above)
Comment (hntd187) [reply src/cluster/rpc_server.rs]: What is a good starting number?

Comment (LucioFranco) [reply src/cluster/rpc_server.rs]: probably in this case you just need 1 thread really.

**Comment by LucioFranco** [comment]
looks pretty good overall, lmk if there is something specific you'd like me to look at.

File: toshi-client/src/lib.rs (hunk 2 above)
Comment (hntd187) [reply toshi-client/src/lib.rs]: I plan to add impls for hyper too through features, this was just easier to get started with an expose to a new lib

**Pull request merged**